src/Microsoft.Azure.SignalR/HubHost/ServiceHubDispatcher.cs

```diff
@@ -65,9 +65,21 @@ public void Start(ConnectionDelegate connectionDelegate, Action<HttpContext> con
             _ = _serviceConnectionManager.StartAsync();
         }
 
-        public Task Shutdown(TimeSpan timeout)
+        private async Task OfflineAndWaitForCompletedAsync()
         {
-            return _serviceConnectionManager.ShutdownAsync(timeout);
+            await Task.WhenAny(
+                _serviceConnectionManager.OfflineAsync(),
+                Task.Delay(TimeSpan.FromSeconds(3))

```

Comment:
provide a cancellation token for the delay

Reply:
and move the timeout check logic into OfflineAsync?

Reply:
Don't feel like to add a `timeout` param for the `OfflineAsync` method.